Sources/FirebaseFirestore/DocumentReference+Swift.swift

```diff
@@ -63,7 +63,10 @@ extension DocumentReference {
           let error = NSError.firestore(errorCode)
           // We only return a snapshot if the error code isn't 0 (aka the 'ok' error code)
           let returned = error == nil ? snapshot?.pointee : nil
-          callback(returned, error)
+
+          DispatchQueue.main.async {
```

Comment:
Could you add a comment here please?  I think that we want to preserve some knowledge of _why_ this is needed (it really does feel like a random change otherwise).

Reply:
yup

Reply:
Done!